src/relay/ir/dataflow_matcher.cc

```diff
@@ -887,6 +895,15 @@ Expr PartitionPattern(DFPattern pattern, Expr expr, Map<String, ObjectRef> attrs
   return PatternPartitioner().Partition(pattern, expr, attrs, check);
 }
 
+Expr PartitionPattern(Array<DFPattern> patterns, Expr expr, Map<String, ObjectRef> attrs,
```

Comment:
I'm not sure, but perhaps this is better expressed as a 'sequence' pattern combinator who's matching rule is what you've written here.